Set the SYSROOT for libxml2 header file to avoid host contamination.

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Yue Tao <yue.tao@windriver.com>
Signed-off-by: Yi Zhao <yi.zhao@windriver.com>

diff --git a/configure.ac.old b/configure.ac
index 54eda73..874f48e 100644
--- a/configure.ac.old
+++ b/configure.ac
@@ -964,15 +964,15 @@ if test "x$squid_opt_use_esi" = "xyes" -a "x$with_libxml2" != "xno" ; then
   dnl Find the main header and include path...
   AC_CACHE_CHECK([location of libxml2 include files], [ac_cv_libxml2_include], [
     AC_CHECK_HEADERS([libxml/parser.h], [], [
-      AC_MSG_NOTICE([Testing in /usr/include/libxml2])
+      AC_MSG_NOTICE([Testing in $SYSROOT/usr/include/libxml2])
       SAVED_CPPFLAGS="$CPPFLAGS"
-      CPPFLAGS="-I/usr/include/libxml2 $CPPFLAGS"
+      CPPFLAGS="-I$SYSROOT/usr/include/libxml2 $CPPFLAGS"
       unset ac_cv_header_libxml_parser_h
-      AC_CHECK_HEADERS([libxml/parser.h], [ac_cv_libxml2_include="-I/usr/include/libxml2"], [
-        AC_MSG_NOTICE([Testing in /usr/local/include/libxml2])
-        CPPFLAGS="-I/usr/local/include/libxml2 $SAVED_CPPFLAGS"
+      AC_CHECK_HEADERS([libxml/parser.h], [ac_cv_libxml2_include="-I$SYSROOT/usr/include/libxml2"], [
+        AC_MSG_NOTICE([Testing in $SYSROOT/usr/local/include/libxml2])
+        CPPFLAGS="-I$SYSROOT/usr/local/include/libxml2 $SAVED_CPPFLAGS"
         unset ac_cv_header_libxml_parser_h
-        AC_CHECK_HEADERS([libxml/parser.h], [ac_cv_libxml2_include="-I/usr/local/include/libxml2"], [
+        AC_CHECK_HEADERS([libxml/parser.h], [ac_cv_libxml2_include="-I$SYSROOT/usr/local/include/libxml2"], [
           AC_MSG_NOTICE([Failed to find libxml2 header file libxml/parser.h])
         ])
       ])
